.github/actions/install-python-deps/action.yml

```diff
@@ -0,0 +1,22 @@
+name: "Install Python Dependencies"
+description: "Installs Python Dependencies from cache or from PyPI if cache is not available.
+Uses pyproject.toml and requirements-dev.txt as cache keys."
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Cached Python Dependencies"
+      uses: getsentry/action-setup-venv@v2.1.0
+      id: venv
+      with:
+        python-version: 3.8
+        venv-dir: ${{ github.workspace }}/python_env
+        cache-dependency-path: |
+          tt_metal/python_env/requirements-dev.txt
+          pyproject.toml
+        install-cmd: make python_env/dev PYTHON_ENV=${{ github.workspace }}/python_env -B
```

Comment:
I believe `-B` should be unnecessary here, in both lines. I would give it a try of removing it and seeing if that helps anything. `-B` forces all targets to be made

Reply:
In my testing, `-B` is absolutely necessary for the downstream building of C++ project files. 
I suggest we keep `-B` refactoring to another ticket since the option is integral to how tt-metal is built.